server/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { AuthDto } from './dto/auth.dto';
+import { CreateUserDto } from '../users/dto';
+import { UsersService } from '../users/users.service';
+import { RefreshTokensService } from '../refresh-tokens/refresh-tokens.service';
+import * as bcrypt from 'bcrypt';
+import { authServiceErrorMessages } from './auth.constants';
+import { AuthTokens, JwtPayload } from './types';
+import { JwtService } from '@nestjs/jwt';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UserEntity } from '../users/user.entity';
+import { FindOptionsWhere } from 'typeorm';
+import { v4 as uuidv4 } from 'uuid';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly usersService: UsersService,
+    private readonly refreshTokensService: RefreshTokensService,
+    private readonly jwtService: JwtService,
+    private readonly appConfigService: AppConfigService,
+  ) {}
+
+  async signUp(createUserDto: CreateUserDto): Promise<AuthTokens> {
+    const userExists = await this.usersService
+      .findOne({ email: createUserDto.email })
+      .catch(() => null);
+
+    if (userExists) {
+      throw new ConflictException(authServiceErrorMessages.entityAlreadyExists);
+    }
+
+    const user = await this.usersService.createOne(createUserDto);
+
+    const refreshTokenId = uuidv4();
+    const tokens = await this.generateTokens(user, refreshTokenId);
+    await this.refreshTokensService.createRefreshToken(
+      { id: user.id },
+      { value: tokens.refreshToken, id: refreshTokenId },
+    );
+
+    return { ...tokens };
```

Comment:
We can omit using desctructurisation
```suggestion
    return tokens;
```